packages/snjs/lib/services/sync/account/response_resolver.ts

```diff
@@ -6,6 +7,31 @@ import { ImmutablePayloadCollection } from '@Protocol/collection/payload_collect
 import { ImmutablePayloadCollectionSet } from '@Protocol/collection/collection_set';
 import { CopyPayload } from '@Payloads/generator';
 import { HistoryMap } from '@Lib/services/history/history_map';
+import { PayloadFormat } from '@Lib/protocol/payloads';
+
+/**
+ * Non-encrypted types are items whose values a server must be able to read.
+ * These include server extensions (such as a note history endpoint), and
+ * multi-factor authentication items, which include a secret value that the server
+ * needs to be able to read in order to enforce.
+ */
+export const NonEncryptedTypes = Object.freeze([
+  ContentType.Mfa,
+  ContentType.ServerExtension,
+]);
+
+function filterDisallowedPayloads(payloads: PurePayload[]): PurePayload[] {
+  return payloads.filter((payload) => {
+    const isDecrypted =
+      payload.format === PayloadFormat.DecryptedBareObject ||
+      payload.format === PayloadFormat.DecryptedBase64String;
+    const allowedToBeDecrypted = NonEncryptedTypes.includes(
+      payload.content_type
+    );
+
+    return !isDecrypted || (isDecrypted && allowedToBeDecrypted);
```

Comment:
could be simplified to:

```js
    isEncrypted || NonEncryptedTypes.includes(
      payload.content_type
    )
```

where `isEncrypted` is:

```js
![PayloadFormat.DecryptedBareObject, PayloadFormat.DecryptedBase64String].includes(payload.format)
```

Reply:
True, but I wonder what's easier to read for a human. I try to essentially model the var names to be like sentences.

Reply:
another way to put this suggestion is:

```js
isEncrypted || isAllowedDecrypted
```

(where `isAllowedDecrypted` is the extracted includes check -- note however that with extracting there's no short-circuiting benefit -- but that may be irrelevant here)

So the sentences we're comparing are "not decrypted or (decrypted and allowed [to be] decrypted)" and "encrypted or allowed decrypted".

I think the latter is clearly easier to read.